buildSrc/src/main/groovy/airbyte-docker.gradle

```diff
@@ -0,0 +1,216 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.slf4j.Logger
+
+import java.nio.file.Path
+import java.security.MessageDigest
+import java.nio.file.Paths
+
+@CacheableTask
+abstract class AirbyteDockerTask extends DefaultTask {
+    @Input
+    abstract File rootDir
+
+    @InputFiles
+    abstract FileCollection projectFiles
+
+    @Input
+    abstract Set<String> baseImageHashes
+
+    @Input
+    abstract File projectDir
+
+    @Input
+    String dockerfileName = 'Dockerfile'
+
+    @Input
+    String testDockerfileName = 'Dockerfile.test'
+
+    @OutputFile
+    abstract File idFileOutput
+
+    def buildDockerfile(String scriptPath, String fileName) {
+        if(project.file(fileName).exists()) {
+            def tag = getTaggedImage(projectDir, dockerfileName)
+
+            project.exec {
+                commandLine scriptPath, rootDir.absolutePath, projectDir.absolutePath, dockerfileName, tag, idFileOutput.absolutePath
+            }
+        }
+    }
+
+    @TaskAction
+    def airbyteDocker() {
+        def scriptPath = Paths.get(rootDir.absolutePath, 'tools/bin/build_image.sh').toString()
+        buildDockerfile(scriptPath, dockerfileName)
+        buildDockerfile(scriptPath, testDockerfileName)
+    }
+
+    static String extractLabelValue(String dockerFile, String labelName) {
+        def file = dockerFile instanceof File ? dockerFile : new File(dockerFile)
+        return file.readLines()
+                .grep({ it.startsWith('LABEL') && it.contains(labelName) })
+                .get(0)
+                .split('=')[1]
+    }
+
+    static String extractImageName(String dockerFile) {
+        return extractLabelValue(dockerFile, "io.airbyte.name")
+    }
+
+    static String extractImageVersion(String dockerFile) {
+        return extractLabelValue(dockerFile, "io.airbyte.version")
+    }
+
+    static String getTaggedImage(projectDir, dockerfileName) {
+        return "${extractImageName(Paths.get(projectDir.absolutePath, dockerfileName).toString())}:dev"
+    }
+}
+
+class AirbyteDockerPlugin implements Plugin<Project> {
+
+    static def getBaseTaggedImages(File dockerfile) {
+        def result = [] as Set<String>
+
+        dockerfile.eachLine { line ->
+            if (line.startsWith("FROM ")) {
```

Comment:
If there's a way it's depending on a separate image that isn't caught by `COPY --from` or `FROM`, that's the worst case, because it won't be caught at all.

If it does use `COPY --from` or `FROM` but uses some unexpected syntax, it'd end up adding the wrong string or an empty string which should almost always result in an inability to find a current image hash, which would always require image rebuilding (slow but not too terrible of an outcome).

I can at least change this to assert the the image is not empty.